File: clases/variables_de_inicio.m
classdef variables_de_inicio

    properties
      alfainicio = 20
      b = 10
      ds = 2
      dsta = 2
      fibdens = 1.77
      fmc = 60
      fvc = 0
      niu = 0.14
      NP = 6
      NR = 10
      resdens = 1.2
      rfm = 15
      rim = 15
      TEX = 800
      vel = 1.5
      zfg = 300
      zfm = 1000
      zi = 0
      zig = 0
      zim = -500
    end

    properties (Dependent)
      lambda1
      RW
    end

    methods
      function obj = Empleado(nombre, edad, salario, puesto)
        % Llamar al constructor de la superclase
        obj@Persona(nombre, edad)
        obj.Salario = salario;
        obj.Puesto = puesto;
      end
        % Método get para la propiedad dependiente c
        function value1 = get.lambda1(obj)
          value1 = obj.niu;
        end

        function value2 = get.RW(obj)
          value2 = obj.b / obj.NR;
        end

        function mostrarInfo(obj)
          disp(['alfainicio: ' obj.alfainicio])
          disp(['b: ' obj.b])
          disp(['ds: ' obj.ds])
          disp(['dsta: ' obj.dsta])
          disp(['fibdens: ' obj.fibdens])
          disp(['fmc: ' obj.fmc])
          disp(['fvc: ' obj.fvc])
          disp(['niu: ' obj.niu])
          disp(['NP: ' obj.NP])
          disp(['NR: ' obj.NR])
          disp(['resdens: ' obj.resdens])
          disp(['rfm: ' obj.rfm])
          disp(['rim: ' obj.rim])
          disp(['TEX: ' obj.TEX])
          disp(['vel: ' obj.vel])
          disp(['zfg: ' obj.zfg])
          disp(['zfm: ' obj.zfm])
          disp(['zi: ' obj.zi])
          disp(['zig: ' obj.zig])
          disp(['zim: ' obj.zim])
          disp(['lambda1: ' obj.lambda1])
          disp(['RW: ' obj.RW])
        end
        function excelInfo(obj)
        % Escribir una celda, ya que es más fácil exportar un archivo con formato de celdas
        celda_variables_de_inicio = {};
        celda_variables_de_inicio(1,:) = {'alfainicio', obj.alfainicio};
        celda_variables_de_inicio(2,:) = {'b', obj.b};
        celda_variables_de_inicio(3,:) = {'ds', obj.ds};
        celda_variables_de_inicio(4,:) = {'dsta', obj.dsta};
        celda_variables_de_inicio(5,:) = {'fibdens', obj.fibdens};
        celda_variables_de_inicio(6,:) = {'fmc', obj.fmc};
        celda_variables_de_inicio(7,:) = {'fvc', obj.fvc};
        celda_variables_de_inicio(8,:) = {'niu', obj.niu};
        celda_variables_de_inicio(9,:) = {'NP', obj.NP};
        celda_variables_de_inicio(10,:) = {'NR', obj.NR};
        celda_variables_de_inicio(11,:) = {'resdens', obj.resdens};
        celda_variables_de_inicio(12,:) = {'rfm', obj.rfm};
        celda_variables_de_inicio(13,:) = {'rim', obj.rim};
        celda_variables_de_inicio(14,:) = {'TEX', obj.TEX};
        celda_variables_de_inicio(15,:) = {'vel', obj.vel};
        celda_variables_de_inicio(16,:) = {'zfg', obj.zfg};
        celda_variables_de_inicio(17,:) = {'zfm', obj.zfm};
        celda_variables_de_inicio(18,:) = {'zi', obj.zi};
        celda_variables_de_inicio(19,:) = {'zig', obj.zig};
        celda_variables_de_inicio(20,:) = {'zim', obj.zim};
        celda_variables_de_inicio(21,:) = {'lambda1', obj.lambda1};
        celda_variables_de_inicio(22,:) = {'RW', obj.RW};
        
        % Escribir un archivo con las variables de entrada
        writecell(celda_variables_de_inicio, obj.excel, 'Sheet', 'Datos de entrada', 'Range', 'A1');
        end
    end
end

File: python/variables_de_inicio.py
import pandas as pd
from openpyxl import Workbook, load_workbook
from write_to_excel import write_to_excel

def variables_de_inicio(excel):
    vector_variables_de_inicio = {
        'alfainicio': 20,  # alfainicio: ángulo de bobinado en la coordenada inicial de bobinado (zi) en grados α ϵ [-90; 90].
        'b': 10,           # b: ancho de hilo de fibra en mm.
        'ds': 2,           # ds: incremento para el cálculo a lo largo de los caminos geodésicos en mm.
        'dsta': 2,         # dsta: incremento para el cálculo a lo largo de las rutas de retorno en mm.
        'fibdens': 1.77,   # fibdens: densidad de la fibra en g/cm3.
        'fmc': 60,         # fmc: contenido en masa de fibra.
        'fvc': 0,          # fvc: contenido en volumen de fibra.
        'niu': 0.14,       # niu: Coeficiente de fricción estática µ.
        'NP': 6,           # NP: Número de caminos por hilo (Number of rovings). El número real de caminos será 4NP.
        'NR': 10,          # NR: Número de mechas (Number of rovings).
        'resdens': 1.2,    # resdens: densidad de la resina en g/cm3.
        'rfm': 15,         # rfm: Radio del mandril en el extremo derecho en mm.
        'rim': 15,         # rim: Radio del mandril en el extremo izquierdo en mm.
        'TEX': 800,        # TEX: valor TEX en g/km
        'vel': 1.5,        # vel: Velocidad de la máquina (multiplica el tiempo que tarda cada línea de comando en realizar el movimiento de los ejes).
        'zfg': 300,        # zfg: Coordenada del extremo geodésico derecho del mandril en mm.
        'zfm': 1000,       # zfm: Coordenada del extremo derecho del mandril en mm.
        'zi': 0,           # zi: Coordenada del inicio del devanado en mm.
        'zig': 0,          # zig: Coordenada del extremo geodésico izquierdo del mandril en mm.
        'zim': -500,       # zim: Coordenada del extremo izquierdo del mandril en mm.
    }
    vector_variables_de_inicio['lambda1'] = vector_variables_de_inicio['niu']                           # lambda1: parámetro λ donde λ = cµ y c ϵ [0; 1].
    vector_variables_de_inicio['RW'] = vector_variables_de_inicio['b'] / vector_variables_de_inicio['NR']  # RW: b/NR - ancho de mecha en mm.

    # Convertir el diccionario a una lista de listas para facilitar la exportación a Excel
    celda_variables_de_inicio = [
        ['alfainicio', vector_variables_de_inicio['alfainicio']],
        ['b', vector_variables_de_inicio['b']],
        ['ds', vector_variables_de_inicio['ds']],
        ['dsta', vector_variables_de_inicio['dsta']],
        ['fibdens', vector_variables_de_inicio['fibdens']],
        ['fmc', vector_variables_de_inicio['fmc']],
        ['fvc', vector_variables_de_inicio['fvc']],
        ['niu', vector_variables_de_inicio['niu']],
        ['NP', vector_variables_de_inicio['NP']],
        ['NR', vector_variables_de_inicio['NR']],
        ['resdens', vector_variables_de_inicio['resdens']],
        ['rfm', vector_variables_de_inicio['rfm']],
        ['rim', vector_variables_de_inicio['rim']],
        ['TEX', vector_variables_de_inicio['TEX']],
        ['vel', vector_variables_de_inicio['vel']],
        ['zfg', vector_variables_de_inicio['zfg']],
        ['zfm', vector_variables_de_inicio['zfm']],
        ['zi', vector_variables_de_inicio['zi']],
        ['zig', vector_variables_de_inicio['zig']],
        ['zim', vector_variables_de_inicio['zim']],
        ['lambda1', vector_variables_de_inicio['lambda1']],
        ['RW', vector_variables_de_inicio['RW']]
    ]
    # Convertir la lista de listas a un DataFrame de pandas
    # Escribir el DataFrame a un archivo Excel
    df = pd.DataFrame(celda_variables_de_inicio, columns=['Variable', 'Valor'])
    write_to_excel(excel, 'Datos de entrada', df)

    return vector_variables_de_inicio

    print('Se ejecutó la función "variables_de_inicio"')